client/pages/Profile.tsx

```diff
@@ -1,14 +1,66 @@
-import React from "react";
+import React, {useEffect, useState} from "react";
 import Sidebar from "@/components/sidebar"; // Adjust the path to match where Sidebar is located in your project
+import supabase from "@/components/supabase";
+
+// @ts-ignore
+function Profile(props) {
+    const [email, setEmail] = useState("");
+    const [fname, setFname] = useState("");
+    const [lname, setLname] = useState("");
+    const [session, setSession] = useState(null);
+
+    useEffect(() => {
+        supabase.auth.getSession().then(({ data: { session } }) => {
+            // @ts-ignore
+            setSession(session)
+        })
+
+        const {
+            data: { subscription },
+        } = supabase.auth.onAuthStateChange((_event, session) => {
+            // @ts-ignore
+            setSession(session)
+        })
+
+        return () => subscription.unsubscribe()
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    },[])
+
+    async function fetchUserData() {
+        const {data, error} = await supabase.from('Users')
+            .select(`
+            id,
+            first_name,
+            last_name,
+            email`);
+        if (error) {
+            console.error("Error fetching User data", error);
```

Comment:
we should look into purging all console logs from the program in a future pr. these feel like they could evolve into security concerns if we aren't careful